File: Groupe1/TP1/src/cercle.c
#include <stdio.h>  // Inclusion de la bibliothèque standard pour les entrées/sorties

#define PI 3.14159   // Définition d'une constante pour la valeur de π (Pi)

int main() {   // Début du programme principal
    double rayon, aire, perimetre;   // Déclaration de trois variables pour stocker le rayon, l'aire et le périmètre

    // Demande du rayon à l'utilisateur
    printf("Entrez le rayon du cercle: ");   // Affichage d'un message demandant à l'utilisateur d'entrer le rayon
    scanf("%lf", &rayon);   // Lecture du rayon entré par l'utilisateur et stockage dans la variable 'rayon'

    // Calcul de l'aire et du périmètre
    aire = PI * rayon * rayon;   // Calcul de l'aire du cercle selon la formule: π * rayon²
    perimetre = 2 * PI * rayon;   // Calcul du périmètre du cercle selon la formule: 2 * π * rayon

    // Affichage des résultats
    printf("L'aire du cercle est: %.2lf\n", aire);   // Affiche l'aire du cercle avec deux décimales
    printf("Le périmètre du cercle est: %.2lf\n", perimetre);   // Affiche le périmètre du cercle avec deux décimales

    return 0;   // Indique la fin normale du programme
}

File: Groupe1/TP1/src/cercle.py
import math   # Importation du module math pour accéder à des fonctions mathématiques comme 'pi'

# Demande du rayon à l'utilisateur
rayon = float(input("Entrez le rayon du cercle: "))  
# Demande à l'utilisateur d'entrer une valeur pour le rayon du cercle
# La valeur saisie est convertie en flottant (float) et stockée dans la variable 'rayon'

# Calcul de l'aire et du périmètre
aire = math.pi * rayon * rayon  
# Calcul de l'aire du cercle avec la formule: π * rayon²
# 'math.pi' fournit la valeur de pi depuis le module math

perimetre = 2 * math.pi * rayon  
# Calcul du périmètre du cercle avec la formule: 2 * π * rayon

# Affichage des résultats
print(f"L'aire du cercle est: {aire:.2f}")  
# Affiche l'aire du cercle avec deux décimales
# Le f-string permet d'insérer directement des variables dans la chaîne de caractères

print(f"Le périmètre du cercle est: {perimetre:.2f}")  
# Affiche le périmètre du cercle avec deux décimales, également en utilisant un f-string
